sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientBuilder.java

```diff
@@ -0,0 +1,248 @@
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public CryptographyClientBuilder keyIdentifier(String keyId) {
+        this.keyId = keyId;
+        return this;
+    }
+
+    /**
+     * Sets the identifier of the jsonWebKey from Azure Key Vault to be used for cryptography operations.
+     *
+     * @param jsonWebKey The Json web jsonWebKey to be used for cryptography operations.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public CryptographyClientBuilder jsonWebKey(JsonWebKey jsonWebKey) {
+        this.jsonWebKey = jsonWebKey;
+        return this;
+    }
+
+    /**
+     * Sets the credential to use when authenticating HTTP requests.
+     *
+     * @param credential The credential to use for authenticating HTTP requests.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws NullPointerException if {@code credential} is {@code null}.
+     */
+    public CryptographyClientBuilder credential(TokenCredential credential) {
+        Objects.requireNonNull(credential);
+        this.credential = credential;
+        return this;
+    }
+
+    /**
+     * Sets the logging level for HTTP requests and responses.
+     *
+     * <p>logLevel is optional. If not provided, default value of {@link HttpLogDetailLevel#NONE} is set.</p>
+     *
+     * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws NullPointerException if {@code logLevel} is {@code null}.
+     */
+    public CryptographyClientBuilder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+        Objects.requireNonNull(logLevel);
+        httpLogDetailLevel = logLevel;
+        return this;
+    }
+
+    /**
+     * Adds a policy to the set of existing policies that are executed after {@link CryptographyAsyncClient} and {@link CryptographyClient} required policies.
+     *
+     * @param policy The {@link HttpPipelinePolicy policy} to be added.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws NullPointerException if {@code policy} is {@code null}.
+     */
+    public CryptographyClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy);
+        policies.add(policy);
+        return this;
+    }
+
+    /**
+     * Sets the HTTP client to use for sending and receiving requests to and from the service.
+     *
+     * @param client The HTTP client to use for requests.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws NullPointerException If {@code client} is {@code null}.
```

Comment:
Recent change but HttpClient and HttpPipeline should be nullable as the builder handles that correctly and it is a valid use case. Look at the change in AppConfiguration for guidance.

No strong need to change this now though

Reply:
will change in P3.